src/ng/compile.js

```diff
@@ -1983,9 +1983,10 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
           var scopeDirective = newIsolateScopeDirective || newScopeDirective;
           var bindings;
           var controllerForBindings;
+          var scopeController;
           if (scopeDirective && elementControllers[scopeDirective.name]) {
             bindings = scopeDirective.$$bindings.bindToController;
-            controller = elementControllers[scopeDirective.name];
+            controller = scopeController = elementControllers[scopeDirective.name];
```

Comment:
I don't know what else scopeDirective can be, so scopeController might not be the best name.